src/testRunner/unittests/tsserver/typingsInstaller.ts

```diff
@@ -1375,6 +1375,33 @@ namespace ts.projectSystem {
             assert.deepEqual(result.newTypingNames, ["jquery", "chroma-js"]);
         });
 
+        it("should not infer typings from filenames with inferTypings:false ", () => {

```

Comment:
You might want to add a unit test that tests setting the new setting on either the project itself or on the host, rather than just in a direct call to discover typings, if possible.